ffc/formatting.py

```diff
@@ -80,6 +80,11 @@ def format_code(code, wrapper_code, prefix, parameters):
     code_h_pre += FORMAT_TEMPLATE["header_h"]
     code_c_pre += FORMAT_TEMPLATE["header_c"]
 
+    # Define ufc_scalar before including ufc.h
+    scalar_type = _define_scalar(parameters)
+    code_h_pre += scalar_type
+    code_c_pre += scalar_type
```

Comment:
Why does the c file have the definition again? It does not include the header file?

Reply:
The header (h file) is not included in the c file. 
I don't know if there is a reason for it, that's why I included the scalar definition on both c an h files. 
But it works both ways.

Reply:
The `.c` does not include the `.h` file to avoid mixing C/C++. The `.h` needs to be valid C/C++, while the `.c` file need to be C. 

C is not a strict subset of C++, so it is good to keep things separate as far as is possible.

Reply:
Ok, so `.c` file is C implementation and `.h` file is a C/C++ layer on top of it?

Reply:
More-or-less.

It is also for `cffi` that I made the split. There is an issue with includes and `cffi`. Let's avoid including the `.h` file in the `.c` file for as long as we can.